source/slang/slang-parameter-binding.cpp

```diff
@@ -1190,20 +1190,31 @@ static void addExplicitParameterBindings_GLSL(
     // If inference is not enabled for this kind, we can issue a warning
     if (!hlslToVulkanLayoutOptions->canInfer(vulkanKind, hlslInfo.space))
     {
-        _maybeDiagnoseMissingVulkanLayoutModifier(context, varDecl.as<VarDeclBase>());
-        return;
+        if(!warnedMissingVulkanLayoutModifier)
+            _maybeDiagnoseMissingVulkanLayoutModifier(context, varDecl.as<VarDeclBase>());
     }
 
     // We use the HLSL binding directly (even though this notionally for GLSL/Vulkan)
     // We'll do the shifting at later later point in _maybeApplyHLSLToVulkanShifts
+    //resInfo = typeLayout->findOrAddResourceInfo(hlslInfo.kind);
     resInfo = typeLayout->findOrAddResourceInfo(hlslInfo.kind);
-    
+
     semanticInfo.kind = resInfo->kind;
     semanticInfo.index = UInt(hlslInfo.index);
     semanticInfo.space = UInt(hlslInfo.space);
-
+    
+    if (resInfo->count.isFinite() && resInfo->count.getFiniteValue() == 0)
```

Comment:
I don't understand what the following code is doing. Why do we set DescriptorTableSlot count to 0?

Reply:
I don't know if this is currently the best way todo this, I will need to look around more

From what I understand, `DescriptorTableSlot` as a representation does not make sense if we are just mapping 1:1 `register(binding,set)`.

Reply:
`DescriptorTableSlot` is `binding`. We rely on it to report the `binding` value for vulkan targets, so the DescirptorTableSlot offset should be exactly the same as the explicit binding value.

Reply:
any reason we use `DescriptorTableSlot` as a binding offset?

Reply:
We don't have any other dedicated enum to represent `the binding offset inside a descriptor set`. We use `DescriptorTableSlot` to mean exactly that for Vulkan.

Reply:
👍we store data same to how a user would use descriptor sets, that makes sense.